### DISCUSSION TITLE
does langserve support langraph natively?

### DISCUSSION THREAD

**Discussion by SoulEvill** [langchain-ai/langserve#586] [General]
Langraph does seem to be an extension of langchain and can be applied to solve new usecases but I have not seen any example that serve langraph application on langserve - does langserve support langgraph natively?

I did a quick test to put together a example from langraph (https://github.com/langchain-ai/langgraph/blob/main/examples/agent_executor/base.ipynb) and tried to host it with langserve. It was able to start the serve without error, but it seems to interpret input schema wrong - instead of only required for input - it recognized required for input and agent_outcome.

**Comment by eyurtsev** [comment]
If you're using langgraph just as a runnable it should support it. The main issue is serialization as LangServe does not currently know of all the messages that one might want to serialize. You can add a last step to the runnable to map the response into something that's json serializable.

**Comment by SoulEvill** [comment]
I see, one of another issue is inferred input schema from langserve wasn't right but I think I can overwrite with specified input shcema. Will try to map response to make it json serializable. let me try it out and report it back!

At the same time - Just curious will there are be more support to langraph on langserve? I am thinking to switch my langchain endpoint to langraph to be more robust (adding retry, circle back all these things)
